manifests/stub.pp

```diff
@@ -11,8 +11,12 @@
   if ! $address {
     fail('unbound::stub: address(es) must be specified.')
   }
-  unbound::stub::validate_addr{ "${address}_${name}":
-    address => $address,
+
+  # An ugly construction that alows us to validate $address in whatever way it
+  # is supplied.
+  $validatable_addresses = suffix(flatten([$address]), ",${name}")
```

Comment:
To make sure all the unbound::stub::validate_addr resources have a unique name.

If we have multiple stub zones with the same addresses, as such:
```puppet
unbound::stub {
  'example.com':
    address => ['10.19.0.3', '10.19.1.3' ];
  'example.net':
    address => ['10.19.0.3', '10.19.1.3' ];
}
```

this will result in

```puppet
$validatable_addresses = [
  '10.19.0.3,example.com',
  '10.19.1.3,example.com',
  '10.19.0.3,example.net',
  '10.19.1.3,example.net'
]
```